lib/routes/favorites_picker.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import './home.dart';
+import '../utils/all_locations.dart';
+import '../utils/favorite_locations.dart';
+
+class FavoritesPicker extends StatefulWidget {
+  const FavoritesPicker({super.key});
+
+  static const String routeName = 'favorites_picker';
+
+  @override
+  State<FavoritesPicker> createState() => _SetFavoriteLocationsState();
+}
+
+class _SetFavoriteLocationsState extends State<FavoritesPicker> {
+  @override
+  Widget build(BuildContext context) {
+    return NotificationListener<ScrollNotification>(
+      onNotification: (notification) {
+        // Returns to the homepage when the user overscrolls
+        if (notification.metrics.pixels >
+                notification.metrics.maxScrollExtent + 100 &&
+            mounted) {
```

Comment:
Nit:
```suggestion
        if (notification.metrics.pixels > notification.metrics.maxScrollExtent + 100
          && mounted) {
```

Reply:
I tried to keep it in the same line but it exceeds the 80 character limit and it gets auto-formatted to the way it was, should I try to change the limit?